safetrace-app/Features/Common/WebViewController.swift

```diff
@@ -221,10 +245,29 @@ extension WebViewController: WKScriptMessageHandler {
 
 extension WebViewController: WKNavigationDelegate, WKUIDelegate {
     func webView(_ webView: WKWebView, didFail navigation: WKNavigation!, withError error: Error) {
-        self.requestWebPage() // retry
+        let error = error as NSError
+        if error.code == NSURLErrorCancelled {
+            // Happens when a resource load was cancelled. This does not indicate that the page failed to load
+            return
+        }
+        print("Error: Failed to request URL for web view. \(error) ")
+
+        showErrorOverlay(retryUrl: webView.url, errorCode: error.code, errorDomain: error.domain)
+    }
+
+    func webView(_ webView: WKWebView, didFailProvisionalNavigation navigation: WKNavigation!, withError error: Error) {
+        let error = error as NSError
+        if error.code == NSURLErrorCancelled {
+            // Happens when another request is made before the previous request is completed. This does not indicate that the page failed to load
+            return
+        }
+        print(error)
```

Comment:
did you mean to keep the print?

Reply:
I did want to surface these somewhere – these were translated from `SLog.print` from Citizen. Though I'm realizing now that it's not the best for this print statement to go into a release build. I can remove these for now

Reply:
Added an `SLog` in this codebase too, which basically just compiles the print statements out of the release builds